drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -87,14 +87,21 @@ local function convert_huesat_st_to_matter(val)
   return utils.clamp_value(math.floor((val * 0xFE) / 100.0 + 0.5), CURRENT_HUESAT_ATTR_MIN, CURRENT_HUESAT_ATTR_MAX)
 end
 
-local function mired_to_kelvin(value)
+local function mired_to_kelvin(value, minOrMax)
   if value == 0 then -- shouldn't happen, but has
     value = 1
     log.warn(string.format("Received a color temperature of 0 mireds. Using a color temperature of 1 mired to avoid divide by zero"))
   end
-  -- we divide inside the rounding and multiply outside of it because we expect these
-  -- bounds to be multiples of 100
-  return utils.round((MIRED_KELVIN_CONVERSION_CONSTANT / value) / 100) * 100
+  -- We divide inside the rounding and multiply outside of it because we expect these
+  -- bounds to be multiples of 100. For the maximum mired value (minimum K value),
+  -- add 1 before converting and round up to nearest hundreds. For the minimum mired
+  -- (maximum K value) value, subtract 1 before converting and round down to nearest
+  -- hundreds.
+  if minOrMax == COLOR_TEMP_MIN then
+    return utils.round(((MIRED_KELVIN_CONVERSION_CONSTANT / (value + 1)) / 100) + 0.5) * 100
```

Comment:
```suggestion
    return utils.round((100 * MIRED_KELVIN_CONVERSION_CONSTANT) / (value + 1) ) + 0.5) * 100
```

Reply:
and then I suggest doing a similar thing for the other case

Reply:
also, this comment does not note where the 1 comes from. would you mind adding this to the explanation you provided?

Reply:
Updated as suggested, and also added a comment explaining that the 1 is needed to avoid rounding errors. Consider the case of a value of 357 mireds, which is equal to 2801 K. Mired values are received from the device as integers and so if we simply round up to the nearest hundreds this would be 2900K. However, 357 mireds is the best representation a manufacturer could provide for a bulb that supports a minimum temp of 2800K so it should of course be rounded to 2800K Adding one to it first ensures that we avoid this rounding error (358 mireds is equal to 2793K which we would then round up to 2800K).

Reply:
great, thanks nick!